ui/app/components/clients/history.js

```diff
@@ -2,49 +2,29 @@ import Component from '@glimmer/component';
 import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
-import { isSameMonth, isAfter, isBefore } from 'date-fns';
+import { isAfter, isBefore, isSameMonth, format } from 'date-fns';
 import getStorage from 'vault/lib/token-storage';
-import { ARRAY_OF_MONTHS } from 'core/utils/date-formatters';
-import { dateFormat } from 'core/helpers/date-format';
 import { parseAPITimestamp } from 'core/utils/date-formatters';
 
-const INPUTTED_START_DATE = 'vault:ui-inputted-start-date';
-
+// TODO CMB: change class and file name to Dashboard
 export default class History extends Component {
   @service store;
   @service version;
 
-  arrayOfMonths = ARRAY_OF_MONTHS;
-
   chartLegend = [
     { key: 'entity_clients', label: 'entity clients' },
     { key: 'non_entity_clients', label: 'non-entity clients' },
   ];
 
-  // FOR START DATE EDIT & MODAL //
-  months = Array.from({ length: 12 }, (item, i) => {
-    return new Date(0, i).toLocaleString('en-US', { month: 'long' });
-  });
-  years = Array.from({ length: 5 }, (item, i) => {
-    return new Date().getFullYear() - i;
-  });
-  currentDate = new Date();
-  currentYear = this.currentDate.getFullYear(); // integer of year
-  currentMonth = this.currentDate.getMonth(); // index of month
-
-  @tracked isEditStartMonthOpen = false;
-  @tracked startMonth = null;
-  @tracked startYear = null;
-  @tracked allowedMonthMax = 12;
-  @tracked disabledYear = null;
-
-  // FOR HISTORY COMPONENT //
-
   // RESPONSE
-  @tracked endTimeFromResponse = this.args.model.endTimeFromResponse;
-  @tracked startTimeFromResponse = this.args.model.startTimeFromLicense; // ex: ['2021', 3] is April 2021 (0 indexed)
-  @tracked startTimeRequested = null;
+  @tracked startMonthTimestamp = this.args.model.licenseStartTimestamp; // when user queries, updates to first month object of response
+  @tracked endMonthTimestamp = this.args.model.currentDate; // when user queries, updates to last month object of response
   @tracked queriedActivityResponse = null;
+  // track params sent to /activity request
+  @tracked activityQueryParams = {
+    start: { timestamp: this.args.model.licenseStartTimestamp }, // updates when user edits billing start month
+    end: { timestamp: this.args.model.currentDate }, // updates when user queries end dates via calendar widget
+  };
```

Comment:
I think these defaults that access `this.args` should be moved to the constructor after `super(...arguments)`. Prior to calling `super` `this.args` would be undefined so I'm surprised if it works like this.

Reply:
To expand on this a bit further, because we are extending the `Component` class and `args` is likely defined on that class it is accessible since `startMonthTimestamp` is being defined after. I think it's probably a better practice to set class property defaults inside the constructor to ensure that the property is available and has the expected value.